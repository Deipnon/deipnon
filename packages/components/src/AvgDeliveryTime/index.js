// @flow

import * as React from 'react'
import { IoIosTime, IoMdTime } from 'react-icons/io'
import { css } from '@emotion/core'
import styled from '@emotion/styled'
import theme from '@deipnon/themes/lib'

const LabelStyle = css`
	margin-bottom: 0;
	margin-left: 1rem;
`
const LabelElement = styled('label')(LabelStyle)

const OpenIconStyle = css`
color: ${theme.styles.AvgDeliveryTimeColorOpen};
box-sizing: border-box;
position: absolute;
margin-left: -1rem;
`

const CloseIconStyle = css`
color: ${theme.styles.AvgDeliveryTimeColorClose};
box-sizing: border-box;
position: absolute;
margin-left: -1rem;
`

const OpenIcon = styled(IoMdTime)(OpenIconStyle)
const CloseIcon = styled(IoIosTime)(CloseIconStyle)


type PropsType = {|
	workingHours: React$Node,
	estimatedDelivery: string,
|}

const AvgDeliveryTime = (props: PropsType) => {
	const date = new Date();
	const time = date.getHours()
	const open = props.workingHours[date.getDay()].opensAt.hours
	const close = props.workingHours[date.getDay()].closesAt.hours

	if (open > time) {
		return <LabelElement><CloseIcon/><span>From </span>{open}</LabelElement>
	} else if (open < time && time < close) {
		return <LabelElement><OpenIcon/>{props.estimatedDelivery}</LabelElement>
	} else {
		return <LabelElement><CloseIcon/></LabelElement>
	}
}

export default AvgDeliveryTime
