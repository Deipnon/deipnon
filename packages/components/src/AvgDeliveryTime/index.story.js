import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'

import AvgDeliveryTime from '.'

const stories = storiesOf('AvgDeliveryTime', module)
stories.addDecorator(withKnobs)

class AvgDeliveryTimeSample extends React.PureComponent {
	constructor (props) {
		super(props)
		this.state = {
			estimatedDelivery : "20dk",
			workingHours: [
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
				{
					opensAt : {
						hours: 20,
						minutes: 30
					},
					closesAt : {
						hours: 23,
						minutes: 30
					}
				},
			]
		}
	}

	render () {
		return (
			<AvgDeliveryTime workingHours={this.state.workingHours} estimatedDelivery={this.state.estimatedDelivery} />
		)
	}
}

stories.add('default', () => <AvgDeliveryTimeSample />)
