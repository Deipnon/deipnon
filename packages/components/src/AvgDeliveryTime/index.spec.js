import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-emotion'

import AvgDeliveryTime from '.'

describe('AvgDeliveryTime', () => {
	test('renders correctly', () => {
		const fakeData =
			[{
				opensAt : {
					hours: 20,
					minutes: 30
				},
				closesAt : {
					hours: 23,
					minutes: 30
				}
			}]

		const tree = renderer.create(<AvgDeliveryTime workingHours={fakeData} estimatedDelivery="20dk" ></AvgDeliveryTime>).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
