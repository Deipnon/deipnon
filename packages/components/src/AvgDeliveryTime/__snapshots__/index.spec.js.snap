// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvgDeliveryTime renders correctly 1`] = `
<label
  className="css-nfz16p"
>
  <svg
    className="css-x367fu"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm14 226c0 7.7-6.3 14-14 14h-96c-7.7 0-14-6.3-14-14s6.3-14 14-14h82V128c0-7.7 6.3-14 14-14s14 6.3 14 14v146z"
    />
  </svg>
</label>
`;
